packages/server/api/src/app/ai/chat/ai-mcp-chat.controller.ts

```diff
@@ -0,0 +1,191 @@
+import { FastifyPluginAsyncTypebox } from '@fastify/type-provider-typebox';
+import { getAiProviderLanguageModel } from '@openops/common';
+import { logger } from '@openops/server-shared';
+import {
+  DeleteChatHistoryRequest,
+  NewMessageRequest,
+  OpenChatMCPRequest,
+  OpenChatResponse,
+  openOpsId,
+  PrincipalType,
+} from '@openops/shared';
+import {
+  CoreAssistantMessage,
+  CoreToolMessage,
+  pipeDataStreamToResponse,
+  streamText,
+  TextPart,
+} from 'ai';
+import { StatusCodes } from 'http-status-codes';
+import { encryptUtils } from '../../helper/encryption';
+import { aiConfigService } from '../config/ai-config.service';
+import {
+  createChatContext,
+  deleteChatHistory,
+  generateChatIdForMCP,
+  getChatContext,
+  getChatHistory,
+  MCPChatContext,
+  saveChatHistory,
+} from './ai-chat.service';
+import { getMcpSystemPrompt } from './prompts.service';
+
+export const aiMCPChatController: FastifyPluginAsyncTypebox = async (app) => {
+  app.post(
+    '/open',
```

Comment:
As we discussed on slack, I think it makes more sense to keep it RESTful and check it in the FE, i.e.
```
var chatId = GET / { contextParams }

if (!chatId)
    chatId = POST / { context params }

PUT /:chatId { message }
```

but let's not do this now, let's revisit the structure once we merge the two controllers